frame/support/src/traits.rs

```diff
@@ -2035,6 +2061,14 @@ pub trait Hooks<BlockNumber> {
 	/// The block is being finalized. Implement to have something happen.
 	fn on_finalize(_n: BlockNumber) {}
 
+	/// The block is being finalized. Implement to have something happen using the remaining weight.
```

Comment:
I think you can copy/paste all the doc of `OnIdle` here and also link to it, something like:

"The function can be used inside to implement [`OnIdle`]."

Because pallet user probably want to read doc from this trait.